otx/core/data/action_dataset_adapter.py

```diff
@@ -0,0 +1,197 @@
+        }  # type: dict
+
+        category_list = []  # to check the duplicate case
+        for cvat_data in datumaro_dataset[Subset.TRAINING]:
+            # Making overall categories
+            categories = cvat_data.categories().get(DatumaroAnnotationType.label, None)
+
+            if categories not in category_list:
+                outputs["category_items"].extend(categories.items)
+                outputs["label_groups"].extend(categories.label_groups)
+
+            category_list.append(categories)
+
+            # Reindexing the each label index of multiple video datasets
+            for cvat_data_item in cvat_data:
+                for ann in cvat_data_item.annotations:
+                    ann_name = categories.items[ann.label].name
+                    ann.label = [
+                        i for i, category_item in enumerate(outputs["category_items"]) if category_item.name == ann_name
+                    ][0]
+
+        # Generate label_entity list according to overall categories
+        outputs["label_entities"] = [
+            LabelEntity(name=class_name.name, domain=self.domain, is_empty=False, id=ID(i))
+            for i, class_name in enumerate(outputs["category_items"])
+        ]
+        return outputs
+
+    def convert_to_otx_format(self, datumaro_dataset: dict) -> Tuple[DatasetEntity, LabelSchemaEntity]:
+        """Convert DatumaroDataset to DatasetEntity for Acion tasks."""
+        raise NotImplementedError
+
+
+class ActionClassificationDatasetAdapter(ActionBaseDatasetAdapter):
+    """Action classification adapter inherited by ActionBaseDatasetAdapter and BaseDatasetAdapter."""
+
+    def convert_to_otx_format(self, datumaro_dataset: dict) -> Tuple[DatasetEntity, LabelSchemaEntity]:
+        """Convert DatumaroDataset to DatasetEntity for Acion Classification."""
+        label_information = self._prepare_label_information(datumaro_dataset)
+        label_entities = label_information["label_entities"]
+
+        label_schema = self._generate_default_label_schema(label_entities)
+        dataset_items = []
+        for subset, subset_data in datumaro_dataset.items():
+            for datumaro_items in subset_data:
+                for datumaro_item in datumaro_items:
+                    image = Image(file_path=datumaro_item.media.path)
+                    shapes = []
+                    for ann in datumaro_item.annotations:
+                        # Action Classification
+                        if isinstance(ann, DatumaroLabel):
+                            shapes.append(
+                                Annotation(
+                                    Rectangle.generate_full_box(), labels=[ScoredLabel(label=label_entities[ann.label])]
+                                )
+                            )
+                    meta_item = MetadataItemEntity(
+                        data=VideoMetadata(
+                            video_id=ID(int(datumaro_item.media.path.split("/")[-3].split("_")[-1])),
```

Comment:
I guess, this is a wrong ID usage, also please consider splitting by os.path sep